crates/libcontainer/src/container/init_builder.rs

```diff
@@ -85,6 +87,14 @@ impl InitContainerBuilder {
         } else {
             None
         };
+        let csocketfd = csocketfd.map(|sockfd| match sockfd {
+            Some(sockfd) => {
+                let fd = sockfd.as_raw_fd();
+                forget(sockfd);
+                fd
+            }
+            None => -1,
+        });
```

Comment:
What happens when it returns `-1`?

Reply:
Apparently, we can combine L90-L94 into L81-89.  Is there a reason why you separate them?

Reply:
> What happens when it returns `-1`?

It doesn't seem to make any difference compared to the past.
The previous `setup_console_socket()` would return `Ok(-1)` under certain conditions. Now, under the same conditions, it returns `Ok(None)`.

Reply:
> Apparently, we can combine L90-L94 into L81-89. Is there a reason why you separate them?

There's nothing special. I can make the modification as you suggested.